packages/pluggableWidgets/column-chart-native/src/components/ColumnChart.tsx

```diff
@@ -0,0 +1,328 @@
+import { BarProps } from "victory-bar";
+import { extractStyles } from "@mendix/pluggable-widgets-tools";
+
+import { ColumnChartStyle } from "../ui/Styles";
+import { SortOrderEnum } from "../../typings/ColumnChartProps";
+import { Legend } from "./Legend";
+import { aggregateGridPadding, mapToAxisStyle, mapToGridStyle, mapToColumnStyles } from "../utils/StyleUtils";
+
+export interface ColumnChartProps {
+    name: string;
+    series: ColumnChartSeries[];
+    sortOrder: SortOrderEnum;
+    style: ColumnChartStyle;
+    presentation: string;
+    showLegend: boolean;
+    showLabels: boolean;
+    xAxisLabel?: string;
+    yAxisLabel?: string;
+    warningPrefix?: string;
+}
+
+export interface ColumnChartSeries {
+    dataPoints: ColumnDataPoints;
+    xFormatter?: (xValue: number | Date | string) => string;
+    yFormatter?: (yValue: number | Date | string) => string;
+    name?: string;
+    customColumnStyle?: string;
+}
+
+export type ColumnDataPoints =
+    | Array<ColumnDataPoint<number, number>>
+    | Array<ColumnDataPoint<number, Date>>
+    | Array<ColumnDataPoint<Date, number>>
+    | Array<ColumnDataPoint<Date, Date>>
+    | Array<ColumnDataPoint<string, string>>
+    | Array<ColumnDataPoint<string, number>>
+    | Array<ColumnDataPoint<string, Date>>
+    | Array<ColumnDataPoint<number, string>>
+    | Array<ColumnDataPoint<Date, string>>;
+
+export interface ColumnDataPoint<X extends number | Date | string, Y extends number | Date | string> {
+    x: X;
+    y: Y;
+}
+
+function sortSeriesDataPoints(series: ColumnChartSeries[], sortingOrder: string): ColumnChartSeries[] {
+    const keysSum: { [key: string]: number } = {};
+    series.forEach(({ dataPoints }) => {
+        dataPoints.forEach(({ x, y }) => {
+            const key = x.toString();
+            if (key in keysSum) {
+                keysSum[key] += Number(y);
+            } else {
+                keysSum[key] = Number(y);
+            }
+        });
+    });
+
+    return series.map(seriesItem => {
+        const dataPoints = seriesItem.dataPoints as Array<ColumnDataPoint<string, number>>;
+        seriesItem.dataPoints = Object.keys(keysSum)
+            .sort((key1, key2) => {
+                return sortingOrder === "descending" ? keysSum[key2] - keysSum[key1] : keysSum[key1] - keysSum[key2];
+            })
+            .map(key => ({
+                x: key,
+                y: dataPoints.find((point: ColumnDataPoint<string, number>) => point.x === key)?.y || 0
+            }));
+
+        return seriesItem;
+    });
+}
+
+export function ColumnChart({
+    name,
+    presentation,
+    series,
+    showLabels,
+    xAxisLabel,
+    yAxisLabel,
+    showLegend,
+    sortOrder,
+    style,
+    warningPrefix
+}: ColumnChartProps): ReactElement | null {
+    const sortedSeries = useMemo(() => {
+        const isNotSortable = series.some(({ dataPoints }) =>
+            dataPoints.some(({ x, y }) => typeof x !== "string" || typeof y !== "number")
```

Comment:
the library itself doesn't allow sorting numbers and dates keys

Reply:
https://formidable.com/open-source/victory/docs/victory-bar/

Reply:
But you're now pre-sorting the data here instead of letting the library handle the sorting, right? If so, does it still matter what the library allows?

"We don't need this feature for numbers and dates" is also a valid answer, btw :P.

Reply:
But you're now pre-sorting the data here instead of letting the library handle the sorting, right? If so, does it still matter what the library allows?

"We don't need this feature for numbers and dates" is also a valid answer, btw :P.

Reply:
if I sorted the keys in respect to their values the result will be something like this [{x: 10, y:1}, {x: 2, y: 3}, {x:4, y: 5}]
if the data is in a stack, the resulted chart will be missing a lot of data. this is a library issue

Reply:
if the numeric or date keys are in correct order, it is works fine.

Reply:
(It actually looks like the current implementation handles this correctly, though? :))

Reply:
yes, I have added some code to sort the keys in an ascending order in case the key are numbers or dates to prevent this issue

Reply:
Then I will gladly resolve this one! 😄